transport.go

```diff
@@ -58,37 +60,69 @@ func remapLocalHost(hostPorts []string) {
 	}
 }
 
-func parsePeer(peer string) (protocol, host string) {
-	// If we get a pure host:port, then we assume tchannel.
+// host and port must be specified in peer
+// explicitProtocol is from --protocol flag
+// protocol will be tchannel, grpc, http, https, ftp in valid case
+// boolean will be false in invalid case
+// TODO(pedge): ftp actually valid? there is a test for it
+func parsePeer(explicitProtocol string, peer string) (protocol, host string, valid bool) {
+	// If we get a pure host:port, then we assume tchannel or grpc.
 	if _, _, err := net.SplitHostPort(peer); err == nil && !strings.Contains(peer, "://") {
-		return "tchannel", peer
+		if explicitProtocol == "" {
+			// If no explicit protocol specified, assume tchannel.
+			return "tchannel", peer, true
+		}
+		switch explicitProtocol {
+		case "http", "https", "ftp", "tchannel", "grpc":
```

Comment:
why do we have `ftp` here?

Reply:
This was part of a test that seemed to test that ftp was valid, I wasn't sure on this one.